examples/js/nodes/utils/ColorSpaceNode.js

```diff
@@ -12,7 +12,13 @@ function ColorSpaceNode( input, method ) {
 
 	this.input = input;
 
-	this.method = method || ColorSpaceNode.LINEAR;
+	this.method = method;
+
+	if ( this.method === undefined ) {
+
+		this.method = this.getEncodingMethod( this.input.value.encoding );
```

Comment:
I think that this already included in TextureNode and CubeTextureNode.

Reply:
Yes, the lines I added above are incorrect and don't fix the issue I was trying to fix. But with or without these lines, colorspace conversion is coming out backwards... when `map.encoding === THREE.sRGBEncoding`, I get the following generated GLSL:

```
vec3 diffuseColor = ( vec4( nVu0, 1.0 ) * LinearTosRGB( tex2D( nVu1, vUv ) ) ).xyz;
```

which should be sRGBToLinear.

Reply:
https://github.com/mrdoob/three.js/blob/8f95052230c154cf846b550243e4ae22ee472969/examples/js/nodes/inputs/TextureNode.js#L60
https://github.com/mrdoob/three.js/blob/8f95052230c154cf846b550243e4ae22ee472969/examples/js/nodes/inputs/CubeTextureNode.js#L56

I think that it being sould be:

```javascript
builder.addContext( { input: code, include: builder.isShader( 'vertex' ) } );

this.colorSpace = this.colorSpace || new ColorSpaceNode( this );
this.colorSpace.fromDecoding( builder.getTextureEncodingFromMap( this.value ) );

code = this.colorSpace.build( builder, this.type );

builder.removeContext();
```

```javascript
ColorSpaceNode.prototype.fromDecoding= function ( decoding ) {

	this.method = this.getDecodingMethod ( decoding )[ 0 ];

	return this;

};
```